voc/python/ast.py

```diff
@@ -325,8 +325,79 @@ def visit_Delete(self, node):
 
     @node_visitor
     def visit_Assign(self, node):
-        # Evaluate the value
-        self.visit(node.value)
+        def push_message(node):
+            # load message on stack
+            self.context.load_name('<generator>')
+            self.context.add_opcodes(
+                JavaOpcodes.GETFIELD('org/python/types/Generator', 'message', 'Lorg/python/Object;')
+            )
+
+            # reset message to None after pushing it on stack
+            self.context.load_name('<generator>')
+            self.context.add_opcodes(
+                JavaOpcodes.INVOKEVIRTUAL(
+                    'org/python/types/Generator',
+                    'reset_message',
+                    args=[],
+                    returns='V'
+                )
+            )
+
+        # walk subtree with node.value as root to find any yield node
+        nested_yield_node_found = False
+        for descendant_node in ast.walk(node.value):
```

Comment:
Is there a reason that this is handled as a separate walk, rather than just visiting `node.value` within the existing AST visitor that we're evaluating?